src/gui/sdlmain.cpp

```diff
@@ -755,8 +753,40 @@ SDL_Rect GFX_GetSDLSurfaceSubwindowDims(Bit16u width, Bit16u height)
 	return rect;
 }
 
+static SDL_Window *setup_window_pp(SCREEN_TYPES screen_type, bool resizable)
+{
+	const SDL_Rect available = get_available_area(sdl.draw.width, sdl.draw.height);
+
+	if (!calc_pp_scale(available.w, available.h))
+		return nullptr;
+
```

Comment:
I noticed further down that this function assigns `sdl.window` and also returns the same value.

Up here in this failure case, should we also assign the `nullptr` to `sdl.window`, so these two branches have the same behavior (albeit different results)?

Reply:
Right, I missed that… returning `nullptr` here will result in one of two cases: either outright `E_Exit` or re-creating window using `surface` output - both bad outcomes, that we would prefer to avoid…

I looked up `ppscale` library and calculating pixel-perfect scale can fail only if input to the library does not make sense; most cases are fine (like original pixel-aspect-ratio being > 0.0 or input dimensions being > 0), but there's one (very hard to reproduce) edge-case, that can trigger this code.

When window is resizable we set minimum window size to the dimensions of drawn area (so based on that, theoretically, the bug should never happen) - however in the past, during testing I've seen that some DEs might not respect minimum window size and via some window manipulations (e.g. forced tiling) resize window below minimum size (I've seen this happen with KDE).

I will add asserts to cover obvious cases, that should never happen and add some recovery code to avoid returning this nullptr.

Reply:
In the process, I found a bug in get_available_area function - but as it turns out, we don't need this function any more :)